app/controllers/hosts_controller.rb

```diff
@@ -261,6 +267,18 @@ def power
     process_error :redirect => :back, :error_msg => _("Failed to %{action} %{host}: %{e}") % { :action => _(params[:power_action]), :host => @host, :e => e }
   end
 
+  def get_power_state
+    result = {:id => @host.id, :state => HOST_POWER[:na]}
+    if @host.supports_power?
+      result[:state] = HOST_POWER[@host.supports_power_and_running? ? :on : :off]
```

Comment:
`supports_power_and_running?` may take a while to return - it has a 15 second timeout, and we hit this controller 20 times when loading the hosts page - this may cause the foreman server to be blocked for several minutes processing these requests.